dojo/tools/kubescape/parser.py

```diff
@@ -59,65 +59,62 @@ def get_findings(self, filename, test):
+                            severity = "Info"
+                            mitigation = ""
+                        else:
+                            severity = self.severity_mapper(controlSummary.get("scoreFactor", 0))
+                            # Define mitigation if available
+                            if "mitigation" in controlSummary:
+                                mitigation = controlSummary["mitigation"]
+                            else:
+                                mitigation = ""
+
+                        description = "**Summary:** " + f"The ressource '{resourceid}' has failed the control '{control_name}'." + "\n"
+                        if controlSummary is not None and "description" in controlSummary:
+                            description += "**Description:** " + controlSummary["description"] + "\n"
+
+                        # Define category if available
+                        if controlSummary is not None and "category" in controlSummary and "subCategory" in controlSummary["category"]:
+                            category_name = controlSummary["category"]["name"]
+                            category_subname = controlSummary["category"]["subCategory"]["name"]
+                            category = f"{category_name} > {category_subname}"
+                            description += "**Category:** " + category + "\n"
+                        elif controlSummary is not None and "category" in controlSummary and "name" in controlSummary["category"]:
+                            category = controlSummary["category"]["name"]
+                            description += "**Category:** " + category + "\n"
+
+                        steps_to_reproduce = "The following rules have failed :" + "\n"
+                        steps_to_reproduce += "\t**Rules:** " + str(json.dumps(control["rules"], indent=4)) + "\n"
+                        steps_to_reproduce += "Resource object may contain evidence:" + "\n"
+                        steps_to_reproduce += "\t**Resource object:** " + str(json.dumps(resource["object"], indent=4))
+                        if rule["status"] != "passed":
+                            find = Finding(
+                                title=textwrap.shorten(title, 150),
+                                test=test,
+                                description=description,
+                                mitigation=mitigation,
+                                steps_to_reproduce=steps_to_reproduce,
+                                severity=severity,
+                                component_name=resourceid,
+                                static_finding=True,
+                                dynamic_finding=False,
+                        )
+                            findings.append(find)
+                            if controlID is not None:
+                                find.unsaved_vulnerability_ids = []
+                                find.unsaved_vulnerability_ids.append(controlID)
```

Comment:
Generally clearer to finish defining the Finding before appending it to the `findings` list.

```suggestion
                            )
                            if controlID is not None:
                                find.unsaved_vulnerability_ids = []
                                find.unsaved_vulnerability_ids.append(controlID)
                            findings.append(find)
```